proof/crefine/RISCV64/IpcCancel_C.thy

```diff
@@ -1769,44 +1775,160 @@ lemma tcb_at_1:
   apply (clarsimp simp add: is_aligned_def ctcb_size_bits_def)
   done
 
+crunch (empty_fail) empty_fail[wp]: inReleaseQueue, isSchedulable
+
 lemma isSchedulable_ccorres [corres]:
   "ccorres (\<lambda>r r'. r = to_bool r') ret__unsigned_long_'
-  (tcb_at' tcbPtr) (UNIV \<inter> {s. thread_' s = tcb_ptr_to_ctcb_ptr tcbPtr})  []
-  (isSchedulable tcbPtr) (Call isSchedulable_'proc)"
-sorry (* FIXME RT: isSchedulable_ccorres *)
+     (tcb_at' tcbPtr and no_0_obj') \<lbrace>\<acute>thread = tcb_ptr_to_ctcb_ptr tcbPtr\<rbrace> []
+     (isSchedulable tcbPtr) (Call isSchedulable_'proc)"
+  supply Collect_const[simp del]
+         if_split[split del]
```

Comment:
Incredibly minor, but do we have a style preference between these?
```suggestion
  supply Collect_const[simp del] if_split[split del]
```

Reply:
Reviewing further, we also sometimes see this alternative
```suggestion
  supply Collect_const[simp del]
  supply if_split[split del]
```

Reply:
My thoughts here were that we might like to emphasise that we're doing different sorts of things. Sometimes I add a bunch of things to the `simp` set and then if `if_split[split del]` is right alongside, it's hard to see. So I thought maybe we'd like one line per attribute, or whatever the correct terminology is. But I'm happy to go along with whatever

Reply:
Hmm, now that you point out the different attributes I think I like your approach